examples/LottieExtension.cpp

```diff
@@ -17,78 +17,111 @@
  * SOFTWARE.
  */
 
+#include <vector>
 #include "Common.h"
 #include <thorvg_lottie.h>
 
 /************************************************************************/
 /* Drawing Commands                                                     */
 /************************************************************************/
 
-static unique_ptr<tvg::LottieAnimation> animation;
-static Elm_Transit *transit;
-static bool updated = false;
+#define NUM_PER_ROW 2
+#define NUM_PER_COL 1
+#define SIZE (WIDTH/NUM_PER_ROW)
 
-void tvgUpdateCmds(tvg::Canvas* canvas, tvg::LottieAnimation* animation, float progress)
-{
-    if (updated || !canvas) return;
+static int counter = 0;
+static std::vector<unique_ptr<tvg::LottieAnimation>> animations;
+static std::vector<Elm_Transit*> transitions;
+static tvg::Canvas* canvas;
 
-    //Update animation frame only when it's changed
-    if (animation->frame(animation->totalFrame() * progress) == tvg::Result::Success) {
+void slotCallback(tvg::LottieAnimation* animation)
+{
+    const char* slotJson = R"({"gradient_fill":{"p":{"a":0,"k":[0,0.1,0.1,0.2,1,1,0.1,0.2,0.1,1]}}})";
+    if (animation->override(slotJson) == tvg::Result::Success) {
         canvas->update();
-        updated = true;
+    } else {
+        cout << "Failed to override the slot" << endl;
     }
 }
 
-void tvgDrawCmds(tvg::Canvas* canvas)
+void markerCallback(tvg::LottieAnimation* animation)
 {
-    //Animation Controller
-    animation = tvg::LottieAnimation::gen();
-    auto picture = animation->picture();
+    if (animation->segment("sectionC") != tvg::Result::Success) {
+        cout << "Failed to segment by the marker" << endl;
+    }
+}
 
-    //Background
-    auto shape = tvg::Shape::gen();
-    shape->appendRect(0, 0, WIDTH, HEIGHT);
-    shape->fill(50, 50, 50);
+void lottieDirCallback(const char* name, const char* path, void* data)
+{
+    if (counter >= NUM_PER_ROW * NUM_PER_COL) return;
 
-    if (canvas->push(std::move(shape)) != tvg::Result::Success) return;
+    char buf[PATH_MAX];
+    snprintf(buf, sizeof(buf), "/%s/%s", path, name);
 
-    const char* slotJson = R"({"gradient_fill":{"p":{"a":0,"k":[0,0.1,0.1,0.2,1,1,0.1,0.2,0.1,1]}}})";
+    //Animation Controller
+    auto animation = tvg::LottieAnimation::gen();
+    auto picture = animation->picture();
 
-    if (picture->load(EXAMPLE_DIR"/lottie/extensions/slotsample.json") != tvg::Result::Success) {
+    if (picture->load(buf) != tvg::Result::Success) {
         cout << "Lottie is not supported. Did you enable Lottie Loader?" << endl;
         return;
     }
+    
+    canvas->push(tvg::cast(picture));
```

Comment:
@fall2019 Hello, good point. It's because the Slot overriding would change properties after parsing lottie JSON, and the parser starts from `canvas->push`